apps/wallet/src/ui/app/hooks/useGetCoins.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useRpcClient } from '@mysten/core';
+import { useQuery } from '@tanstack/react-query';
+
+import type { SuiAddress, PaginatedCoins, CoinStruct } from '@mysten/sui.js';
+// minimum number of coins to fetch per request
+const MIN_COINS_PER_REQUEST = 100;
+
+// Fetch all coins for an address, this will keep calling the API until all coins are fetched
+export function useGetCoins(coinType: string, address?: SuiAddress | null) {
```

Comment:
This is probably fine for now, but is actually a non-ideal approach for coin management, since we really should only fetch as many coins as we need to, when we need them, vs fetching all of the individual coins always.

Reply:
This is primarily for transfer-coins and staking. Does this mean we refetch whenever a user changes the amount, and get gas budget and validation?

Reply:
It's a little tricky, but for budgeting and validation we should only need the total balance, and in those cases when we send the transaction we really want the latest up-to-date coins, not to pull from a coin cache. We can punt this refactoring to when we have programmable transactions though.